core/src/main/scala/kafka/raft/KafkaMetadataLog.scala

```diff
@@ -443,11 +443,12 @@ object KafkaMetadataLog {
 
   private def deleteSnapshotFiles(
     logDir: Path,
-    expiredSnapshots: mutable.TreeMap[OffsetAndEpoch, Option[FileRawSnapshotReader]]
+    expiredSnapshots: mutable.TreeMap[OffsetAndEpoch, Option[FileRawSnapshotReader]],
+    logging: Logging
   ): () => Unit = () => {
     expiredSnapshots.foreach { case (snapshotId, snapshotReader) =>
       snapshotReader.foreach { reader =>
-        CoreUtils.swallow(reader.close(), this)
+        CoreUtils.swallow(reader.close(), logging)
```

Comment:
This is an existing issue. But CoreUtils.swallow() doesn't seem to need logging.